pkg/route/monitor/monitor.go

```diff
@@ -0,0 +1,78 @@
+package monitor
+
+import (
+	"sync"
+
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/cache"
+)
+
+type objectKey struct {
+	namespace string
+	name      string
+	uid       types.UID
+}
+
+type Object struct {
+	RefCount int
```

Comment:
This seems more suitable as a property tracked by the caller.

Reply:
Yeah I would agree, will rework this bit.